### DISCUSSION TITLE
Alguém pode me ajudar como executar comandos ao iniciar no i3wm sem pedir a senha?

### DISCUSSION THREAD

**Discussion by ronibastos** [i3/i3#6013] [Q&A]
vim .config/i3/config

exec --no-startup-id alacritty -e sudo apt autoremove
exec --no-startup-id alacritty -e sudo apt autoclean
exec --no-startup-id alacritty -e sudo apt install -f
exec --no-startup-id alacritty -e sudo apt update && -e sudo apt upgrade
exec --no-startup-id alacritty -e sudo apt update
exec --no-startup-id alacritty -e sudo apt upgrade
exec --no-startup-id alacritty -e sudo apt full-upgrade

como executar esses comandos ao iniciar no i3wm sem pedir a senha?

**Comment by pietrobondioli** [comment]
Tudo bem Roni? Estava passando aqui pela wiki do i3 pra checar um tópico e vi sua questão.

Acho que para o seu caso específico faz muito mais sentido usar um systemd service, com ele você consegue executar um comando/script que executa uma vez sempre que o seu sistema se inicia. Você teria algo tipo:

/etc/systemd/system/update_script.service
```
[Unit]
Description=Update script service
After=network.target

[Service]
Type=oneshot
ExecStart=/path/to/update_script.sh

[Install]
WantedBy=multi-user.target
```

E claro, não esqueça de criar no caminho correto o script com os comandos que você quer (além de dar as permissoes corretas pro script, acho que 644 seria o correto), após isso basta iniciar/ativar o serviço. Pela internet você pode achar mais informações sobre como o systemd funciona, a arch wiki é uma boa fonte (https://wiki.archlinux.org/title/Systemd).

**Reply by ronibastos** [reply]
Tudo jóia Pietro
Muito Obrigado pela ajuda e dicas! Vou testar aqui na minha maquina! Valeu!

**Reply by okraits** [reply]
Please use English so that other people can read and learn from the question and answers as well.
